test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,292 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from haystack import Document
+from haystack.components.preprocessors.markdown_header_splitter import MarkdownHeaderSplitter
+
+
+# Fixtures
+@pytest.fixture
+def sample_text():
+    return (
+        "# Header 1\n"
+        "Content under header 1.\n"
+        "## Header 1.1\n"
+        "### Subheader 1.1.1\n"
+        "Content under sub-header 1.1.1\n"
+        "## Header 1.2\n"
+        "### Subheader 1.2.1\n"
+        "Content under header 1.2.1.\n"
+        "### Subheader 1.2.2\n"
+        "Content under header 1.2.2.\n"
+        "### Subheader 1.2.3\n"
+        "Content under header 1.2.3."
+    )
+
+
+# Basic splitting and structure
+def test_basic_split(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Should split into all headers with content
+    headers = [doc.meta["header"] for doc in split_docs]
+    assert "Header 1" in headers
+    assert "Subheader 1.1.1" in headers
+    assert "Subheader 1.2.1" in headers
+    assert "Subheader 1.2.2" in headers
+    assert "Subheader 1.2.3" in headers
+
+    # Check that content is present and correct
+    header1_doc = next(doc for doc in split_docs if doc.meta["header"] == "Header 1")
+    assert "Content under header 1." in header1_doc.content
+
+    subheader111_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.1.1")
+    assert "Content under sub-header 1.1.1" in subheader111_doc.content
+
+    subheader121_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.1")
+    assert "Content under header 1.2.1." in subheader121_doc.content
+
+    subheader122_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.2")
+    assert "Content under header 1.2.2." in subheader122_doc.content
+
+    subheader123_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.3")
+    assert "Content under header 1.2.3." in subheader123_doc.content
+
+    # Ensure all documents have a header in their metadata
+    for doc in split_docs:
+        assert doc.meta.get("header") is not None
+
+
+def test_split_parentheaders(sample_text):
```

Comment:
If we update the `test_basic_split` test following my comment https://github.com/deepset-ai/haystack/pull/9660/files#r2432308570 then we can drop this test.